shoreline.go

```diff
@@ -39,6 +43,30 @@ type (
 	}
 )
 
+func kafkaSender() *kafka_sarama.Sender {
+	prefix, _ := os.LookupEnv("KAFKA_PREFIX")
+	topic, _ := os.LookupEnv("KAFKA_TOPIC")
+	topicWithPrefix := prefix + topic
+	broker, _ := os.LookupEnv("KAFKA_BROKERS")
+	log.Println(broker)
+	log.Println(topicWithPrefix)
+
+	saramaConfig := sarama.NewConfig()
+	saramaConfig.Version = sarama.V2_0_0_0
+
+	sender, err := kafka_sarama.NewSender([]string{broker}, saramaConfig, topicWithPrefix)
+	if err != nil {
+		log.Printf("failed to create protocol: %s", err.Error())
+	}
+	return sender
+}
+func kafkaClient(Sender *kafka_sarama.Sender) (cloudevents.Client) {
+	c, err := cloudevents.NewClient(Sender, cloudevents.WithTimeNow(), cloudevents.WithUUIDs())
+	if err != nil {
+		log.Printf("failed to create client, %v", err)
+	}
+	return c
+}
 func main() {
```

Comment:
I recommend using fx and envconfig to separate concerns.  The first concern is building the separate components needed to wire up your application.  The second concern is to create objects that are initialized from the environment.

Reply:
to implement fx in shoreline at a later date. implemented envconfig.